plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotIntegrationSmokeTest.java

```diff
@@ -0,0 +1,357 @@
+                        "  MIN(double_col), MIN(element_at(double_array_col, 1))" +
+                        "  FROM " + ALL_TYPES +
+                        "  WHERE bytes_col = X''",
+                "VALUES(0, " + Long.MIN_VALUE + ", " +
+                        "  0, " + Integer.MIN_VALUE + ", 7," +
+                        "  0.0, -POWER(0, -1)," +
+                        "  0.0, -POWER(0, -1)" +
+                        ")");
+
+        // Default null value for strings is the string 'null'
+        // Default null value for booleans is the string 'null', boolean is treated as a string
+        assertQuery("SELECT DISTINCT string_col, element_at(string_array_col, 1), bool_col, element_at(bool_array_col, 1)" +
+                        "  FROM " + ALL_TYPES +
+                        "  WHERE bytes_col = X'' and element_at(int_array_col_with_pinot_default, 1) = 7",
+                "VALUES ('null', 'null', 'null', 'null')");
+
+        // Null behavior for arrays:
+        // Default value for a "null" array is 1 element with default null array value,
+        // ex.
+        // Null string or boolean arrays will have ['null'] as the default,
+        // Null integer arrays will have [Integer.MIN_VALUE] as the default
+        // Null long arrays will have [Long.MIN_VALUE] as the default,
+        // Null float arrays will have [-INFINITY] as the default
+        // Null double arrays will have [-INFINITY] as the default
+        // As mentioned above, arrays of varbinary are not supported
+        assertQuery("SELECT CARDINALITY(string_array_col), CARDINALITY(bool_array_col), CARDINALITY(int_array_col),  CARDINALITY(int_array_col_with_pinot_default)," +
+                        "  CARDINALITY(float_array_col),  CARDINALITY(long_array_col),  CARDINALITY(long_array_col)" +
+                        "  FROM " + ALL_TYPES +
+                        "  WHERE bytes_col = X'' and element_at(int_array_col_with_pinot_default, 1) = 7",
+                "VALUES " + join(", ", IntStream.range(0, NULL_COUNT - ARRAY_NULL_COUNT).mapToObj(i -> "(1, 1, 1, 1, 1, 1, 1)").collect(toList())));
+
+        // If an array contains both null and non-null values, the null values are omitted:
+        // There are 5 values in the avro records, but only the 3 non-null values are in pinot
+        assertQuery("SELECT CARDINALITY(string_array_col), CARDINALITY(bool_array_col), CARDINALITY(int_array_col),  CARDINALITY(int_array_col_with_pinot_default)," +
+                        "  CARDINALITY(float_array_col),  CARDINALITY(long_array_col),  CARDINALITY(long_array_col)" +
+                        "  FROM " + ALL_TYPES +
+                        "  WHERE bytes_col = X'' and element_at(bool_array_col, 1) != 'null'",
+                "VALUES " + join(", ", IntStream.range(0, ARRAY_NULL_COUNT).mapToObj(i -> "(3, 3, 1, 3, 1, 1, 1)").collect(toList())));
+    }
+
+    @Test
+    public void testAggregationPushdown()
```

Comment:
seems like there is some overlap between this test and TestMinimalFunctionality.
Maybe having one would be enough?
I'd prefer having tests here, and trying to migrate away of "test minimal functionality" classes (they are lightweight, but then it's unclear which tests should be added there and which ones here, etc.)

Reply:
Removed TestMinimalFunctionality, all the tests were merged into the smoke test.

Reply:
Could Pinot reuse `io.trino.plugin.jdbc.BaseJdbcConnectorTest#testAggregationPushdown`?